src/com/pensio/api/request/ChargeSubscriptionRequest.java

```diff
@@ -2,13 +2,30 @@
 
 import com.pensio.Amount;
 
-public class ChargeSubscriptionRequest
+public class ChargeSubscriptionRequest<T extends ChargeSubscriptionRequest<T>>
 {
 	private String agreementId;
 	private AgreementUnscheduledType agreementUnscheduledType;
 	private Amount amount;
 	private String reconciliationIdentifier;
-	
+	private PaymentInfos paymentInfos;
+
+    {
+        paymentInfos = new PaymentInfos();
+    }
+
+    public PaymentInfos getPaymentInfos()
+    {
+        return paymentInfos;
+    }
+
+    @SuppressWarnings("unchecked")
+    public T addPaymentInfo(String key, String value)
```

Comment:
Why are returning T, if the object is strictly String ?

Reply:
T is used here to support type-safe chaining in derived classes, consistent with similar classes using paymentInfo. This allows addPaymentInfo to return the subclass type rather than just ChargeSubscriptionRequest.